language/bytecode-verifier/bytecode-verifier-tests/src/unit_tests/bounds_tests.rs

```diff
@@ -14,6 +14,13 @@ use move_core_types::{
 };
 use proptest::{collection::vec, prelude::*};
 
+macro_rules! assert_err_code {
+    ($err:expr, $code:expr) => {
+        use StatusCode::*;
+        assert_eq!($err.major_status(), $code);
+    };
+}
```

Comment:
I'm eager to hear feedback on using a macro here. I feel like the `use` saves some typing, and using a macro ensures the inner `assert_eq!` points to the right spot should the test that uses the macro fails. But if someone told me using a macro here is ugly, I'd agree with that sentiment as well 😅

Reply:
Opinions on this definitely vary! I personally like any macros that make tests more convenient to write, but am significantly more wary of macro usage in "real" code. Interested to hear the philosophy of @tzakian and @tnowacki, who I think are more pro-macro than me.

Reply:
I would probably just prefer to see this macro inlined. I think `assert_eq(m.freeze().unwrap_err().major_status(), ... error code...)` is a bit more desirable as it is obvious that other aspects of the error are not being checked. 

So please keep all these additional asserts! Just maybe ditch the new macro